jekyll/_cci2/configuration-reference.md

```diff
@@ -1391,6 +1391,181 @@ ignore | N | String, or List of Strings | Either a single tag specifier, or a li
+            # The job will be invoked once for each combination of parameters.
+            # In this case, the job would be invoked 9 times, once for each
+            # version/platform combination.
+            parameters:
+              version: ["0.1", "0.2", "1.0"]
+              platform: ["windows", "linux", "macos"]
+
+            # You may provide a matrix alias in order to require the entire
+            # matrix later. The default alias is the name of the job being
+            # invoked, which in this case would be:
+            # alias: build
+            alias: my-matrix
+
+            # You may `exclude` some sets of parameters from the matrix.
+            exclude:
+              - version: 0.2
+                platform: linux
+              - version: 1.0
+                platform: macos
+```
+
+###### Matrix parameters
+{:.no_toc}
+When running a job in a matrix, other job arguments (such as `requires` or
+`name`) can contain matrix parameters, in the form `<< matrix.[parameter-name] >>`
+For example:
+```
+- deploy:
+    matrix:
+      parameters:
+        version: ["0.1", "0.2"]
+    name: "deploy-<< matrix.version >>"
+    requires:
+      - "build-<< matrix.version >>"
+```
+
+This matrix result in two jobs being run. One, called `deploy-0.1`,
+would require the `build-0.1` job. The other, called `deploy-0.2`, would require
+the `build-0.2` job.
+
+###### Excluding sets of parameters from a matrix
+{:.no_toc}
+Sometimes you may wish to run a job with every combination of arguments _except_
+some value or values. You can use an `exclude` stanza to achieve this:
+
+```
+- build:
+    matrix:
+      parameters:
+        a: [1, 2, 3]
+        b: [4, 5, 6]
+      exclude:
+        - a: 3
+          b: 5
+```
+
+The matrix above would result in 8 jobs being run: every combination of the
+parameters `a` and `b`, excluding `{a: 3, b: 5}`
+
+###### Requiring an entire matrix
+{:.no_toc}
+Another job can `require` an entire matrix (that is, every job in the matrix)
+using its `alias`. The `alias` defaults to the name of the job being invoked.
+```
+- deploy:
+    matrix:
+      parameters:
+        version: ["0.1", "0.2"]
+- another-job:
+    requires:
+      - deploy
+```
+
+This means that `another-job` will require both `deploy-0.1` and `deploy-0.2` to
+finish before it runs. 
+
+You can also manually set the `alias` to a custom value. For example:
+
+```
+- deploy:
+    matrix:
+      parameters:
+        version: ["0.1", "0.2"]
+      # setting a custom alias
+      alias: my-deploy-matrix
+- another-job:
+    requires:
+      # using a custom alias
+      - my-deploy-matrix
+```
+
+###### Requiring a single job from a matrix
+{:.no_toc}
+Every job in a matrix has a name. By default, the name will be the job's name,
+concatenated with the value of each matrix parameter alphabetized by the
+*parameter name*, separated by hyphens. These names can be used to `require` a
+job, including from within another matrix:
```

Comment:
Going into depth on how the default name is generated feels a bit TMI to me, since it's meant to be specified by the user if they have a specific opinion on what it should look like.

Reply:
I was thinking that it'd be more comprehensible if I could explain how to require a job _without_ specifically naming it... but that seems silly. I removed this part and just explicitly named the job and required that.